src/anyp/Uri.h

```diff
@@ -35,17 +35,19 @@ class Uri
     Uri(): hostIsNumeric_(false) { *host_ = 0; }
     Uri(AnyP::UriScheme const &aScheme);
     Uri(const Uri &other) {
-        this->operator =(other);
-    }
-    Uri &operator =(const Uri &o) {
-        scheme_ = o.scheme_;
-        userInfo_ = o.userInfo_;
-        memcpy(host_, o.host_, sizeof(host_));
-        hostIsNumeric_ = o.hostIsNumeric_;
-        hostAddr_ = o.hostAddr_;
-        port_ = o.port_;
-        path_ = o.path_;
+        scheme_ = other.scheme_;
+        userInfo_ = other.userInfo_;
+        memcpy(host_, other.host_, sizeof(host_));
```

Comment:
Would it be accurate to say that the official assignment operator implementation is correct except that this memcpy() should be memmove()?

N.B. This comment does not request any changes. I am just trying to understand the true scope of the problem.

Reply:
I am interpreting your answer as "yes, memcpy() is the only bug here".

> AFAIK adding the missing if-statement for self-assign check will optimize away...

As for an optimization opportunity, when self-assignments are very rare, adding _unnecessary_ `if` statements is the _opposite_ of optimization: In such contexts, it is overall better to (safely) copy in all cases, then to pay the price of avoiding unnecessary (but exceptionally rare) copies. In most contexts, self-assignments are very rare.